pkg/operator/ceph/object/bucket/util.go

```diff
@@ -93,44 +94,65 @@ func additionalConfigSpecFromMap(config map[string]string) (*additionalConfigSpe
 	spec := additionalConfigSpec{}
 
 	if _, ok := config["maxObjects"]; ok {
+		if !opcontroller.ObcAdditionalConfigKeyIsAllowed("maxObjects") {
```

Comment:
General design question: what is preferred approach to represent such options in Rook CRDs?

There is some examples, where `map[string]stirng` is used to represent options. Having map in spec would allow to simplify multiple if statements to a single loop. But having struct in spec provides a more strict interface to users.

Reply:
For OBC's the `additionalConfig` `map[string]string` (an opaque parameter map) is the interface available to us, so that's why this is used here.

In Rook generally, I prefer typed fields except when a CR is acting as a straightforward passthrough. For example, the CephBlockPool `parameters` opaque map is a passthru for arbitrary Ceph block pool options.

Another consideration is whether a config requires a daemon restart or not, or if a config requires some other special handling. Those configs are good candidates for typed fields. 

Over time, I've grown to like passthru options as a way of allowing users more flexibility over tunables, and also allowing Rook maintainers to focus on high-impact features instead of chasing arbitrary, small-impact configs.